### DISCUSSION TITLE
Predicting insta-mineable blocks

### DISCUSSION THREAD

**Discussion by Pixlent** [Minestom/Minestom#1954] [Help]
Hi! I'm making a custom mining system which has never been done before by anyone, ever, but there's an issue. Instant mining isn't really instant and before I didn't really have a solution, but the server "Origin Realms" somewhat found a solution. I was randomly on the server and found out you instantly mine blocks, almost like vanilla. So far my hypothesis is that they predict where the player will mine (aka get the block behind the block currently being mined and break it).

I want to implement this solution, but I'm afraid I don't know how to do raycasting (lol). I'm trying to raycast the blocks in front of the player, ignoring the block the player is currently mining and having a distance of about ~4 blocks, how would I achieve this? I've tried a for loop, rayfast (emortal, grid casting) but so far nothing has worked. Would anyone be able to assist?

**Comment by x7airworker** [comment]
There is https://javadoc.minestom.net/net/minestom/server/entity/LivingEntity.html#getTargetBlockPosition(int) which I've used before. I don't know how accurate this method is, but I didn't have any problems with this method.
